src/memory_pool_default.c

```diff
@@ -36,7 +45,8 @@ umf_result_t umfPoolCreateInternal(const umf_memory_pool_ops_t *ops,
     pool->ops = *ops;
     ret = ops->initialize(pool->provider, params, &pool->pool_priv);
     if (ret != UMF_RESULT_SUCCESS) {
-        free(pool);
+        umf_ba_free(BA_pool, pool);
+        umfBaseAllocatorDestroy();
```

Comment:
This is because we don't want to wait until library dtor to free the memory. For critnib it's  fine since there is only one instance of critnib. However, as I mentioned in a previous comment, we could still move it to library dtor if only ba_alloc would not keep coarse-grain allocations until destroy is called.

Reply:
If I'll have time I will propose a change to ba_alloc to address this.

Reply:
Done